packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -1149,18 +1162,58 @@ export class CdkToolkit {
     return selectedForDiff;
   }
 
-  private async selectStacksForDestroy(selector: StackSelector, exclusively?: boolean) {
-    const assembly = await this.assembly();
+  private async selectStacksForDestroy(selector: StackSelector, assembly: CloudAssembly, exclusively?: boolean) {
```

Comment:
Please keep the pattern of creating the assembly on the fly for now. It's cached by default, so doesn't add much overhead.

```suggestion
  private async selectStacksForDestroy(selector: StackSelector, exclusively?: boolean) {
    const assembly = await this.assembly();
```

It's a good idea though and I have a related refactor in the pipeline! But for now I'd like to keep it consistent.